src/NSwag.SwaggerGeneration.AspNetCore/AspNetCoreToSwaggerGenerator.cs

```diff
@@ -145,14 +145,45 @@ public async Task<SwaggerDocument> GenerateAsync(object serviceProvider)
                     }
                 };
 
-                operations.Add(new Tuple<SwaggerOperationDescription, ApiDescription, MethodInfo>(operationDescription, apiDescription, method));
+                var consumes = apiDescription.SupportedRequestFormats
+                   .Select(f => f.MediaType)
+                   .Distinct();
+
+                var produces = apiDescription.SupportedResponseTypes
+                   .SelectMany(t => t.ApiResponseFormats.Select(f => f.MediaType))
+                   .Distinct();
+
+                operations.Add(new Tuple<SwaggerOperationDescription, ApiDescription, MethodInfo, IEnumerable<string>, IEnumerable<string>>(
+                    operationDescription, apiDescription, method, consumes, produces));
             }
 
             return await AddOperationDescriptionsToDocumentAsync(document, controllerType, operations, swaggerGenerator, schemaResolver).ConfigureAwait(false);
         }
 
-        private async Task<bool> AddOperationDescriptionsToDocumentAsync(SwaggerDocument document, Type controllerType, List<Tuple<SwaggerOperationDescription, ApiDescription, MethodInfo>> operations, SwaggerGenerator swaggerGenerator, SwaggerSchemaResolver schemaResolver)
+        private async Task<bool> AddOperationDescriptionsToDocumentAsync(SwaggerDocument document, Type controllerType,
+            List<Tuple<SwaggerOperationDescription, ApiDescription, MethodInfo, IEnumerable<string>, IEnumerable<string>>> operations,
+            SwaggerGenerator swaggerGenerator, SwaggerSchemaResolver schemaResolver)
         {
+            var globalConsumes = operations
+                .SelectMany(o => o.Item4)
+                .Where(c => operations.All(o => o.Item4.Count() == 0 || o.Item4.Contains(c)))
```

Comment:
Super nit: Might be useful to start using `ValueTuple` so you could start doing `List<(SwaggerOperationDescription operationDescription, ApiDescription apiDescription, MethodInfo method, IEnumerable<string> consumes, IEnumerable<string> produces)>`. Makes it far more readable

Reply:
Yes, I dont like this either - even using ValueTuple too much duplicated code for me :-) maybe an internal class is even better... But the bigger question is whether the new logic is correct :-). Thanks.